api/src/services/Datastream.test.ts

```diff
@@ -0,0 +1,95 @@
+import FedoraCatalog from "./FedoraCatalog";
+import { FedoraObject } from "../models/FedoraObject";
+import Datastream from "./Datastream";
+import Config from "../models/Config";
+
+describe("Datastream", () => {
+    let updateDatastreamFromFileSpy;
+    let datastream;
+    let datastreamsSpy;
+    beforeEach(() => {
+        jest.spyOn(Config, "getInstance").mockReturnValue(null);
+        updateDatastreamFromFileSpy = jest.spyOn(FedoraObject.prototype, "updateDatastreamFromFile");
+        datastreamsSpy = jest.spyOn(FedoraCatalog.prototype, "getDatastreamMimetypes");
+        datastream = Datastream.getInstance();
+    });
+
+    describe("hasValidMimeType", () => {
+        beforeEach(() => {
+            datastreamsSpy.mockReturnValue({
+                WAV: {
+                    mimetype: {
+                        allowedType: "test1",
+                        allowedSubtypes: "test2,test3,test4",
+                    },
+                },
+                THUMBNAIL: {
+                    mimetype: {
+                        allowedType: "*",
+                        allowedSubtypes: "test2,test3,test4",
+                    },
+                },
+                MEDIUM: {
+                    mimetype: {
+                        allowedType: "test1",
+                        allowedSubtypes: "*",
+                    },
+                },
+            });
+        });
+
+        it("returns false when subtype does not exist", () => {
+            expect(datastream.hasValidMimeType("WAV", "test1/test5")).toBeFalsy();
+        });
+
+        it("returns false when type does not exist", () => {
+            expect(datastream.hasValidMimeType("WAV", "test5/test2")).toBeFalsy();
+        });
+
+        it("returns true when type/subtype does exist", () => {
+            expect(datastream.hasValidMimeType("WAV", "test1/test2")).toBeTruthy();
+        });
+
+        it("returns true for any type", () => {
```

Comment:
This appears to be a duplicate test case (combining code from above and name from below).

Reply:
made change